File: license-key-formatting/license_key_formatting.py
class Solution:
    def licenseKeyFormatting(self, S: 'str', K: 'int') -> 'str':
        a = ''.join(S.split('-'))[::-1]
        res = ''
        i = 0
        for c in a:
            if i != K:
                res += c.upper()
                i += 1
            else:
                res += '-' + c.upper()
                i = 1
        return res[::-1]

File: license-key-formatting/license_key_formatting.go
func licenseKeyFormatting(S string, K int) string {
	runes := make([]rune, 0)
	for _, c := range S {
		if c != '-' {
			runes = append(runes, c)
		}
	}
	res := make([]rune, 0)
	i := 0
	for _, c := range reverse(runes) {
		if i != K {
			res = append(res, c)
			i++
		} else {
			res = append(res, '-')
			res = append(res, c)
			i = 1
		}
	}
	return reverse(toupper(res))
}

func reverse(runes []rune) string {
	for from, to := 0, len(runes)-1; from < to; from, to = from+1, to-1 {
		runes[from], runes[to] = runes[to], runes[from]
	}
	return string(runes)
}

func toupper(runes []rune) []rune {
	for i := 0; i < len(runes); i++ {
		if runes[i] >= 97 && runes[i] <= 122 {
			runes[i] -= 32
		}
	}
	return runes
}
